apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/AllowSamhsaFilter.java

```diff
@@ -0,0 +1,99 @@
+package gov.cms.bfd.server.war;
+
+import static gov.cms.bfd.server.war.SpringConfiguration.SSM_PATH_SAMHSA_ALLOWED_CERTIFICATE_ALIASES_JSON;
+import static gov.cms.bfd.server.war.commons.CommonTransformerUtils.SHOULD_FILTER_SAMHSA;
+
+import com.google.gson.Gson;
+import gov.cms.bfd.server.war.commons.ClientCertificateUtils;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/***
+ * test.
+ */
+@Component("AllowSamhsaFilterBean")
+public class AllowSamhsaFilter extends OncePerRequestFilter {
+
+  /** The logger. */
+  private static final Logger LOGGER = LoggerFactory.getLogger(AllowSamhsaFilter.class);
+
+  /** List of allowed certificate serial numbers. */
+  private final List<BigInteger> samhsaAllowedSerialNumbers;
+
+  /**
+   * Creates a new {@link AllowSamhsaFilter}.
+   *
+   * @param samhsaAllowedCertificateAliasesJson list of certificate aliases to identify clients that
+   *     are allowed to see SAMHSA data
+   * @param keyStore server key store
+   */
+  public AllowSamhsaFilter(
+      @Value("${" + SSM_PATH_SAMHSA_ALLOWED_CERTIFICATE_ALIASES_JSON + "}")
+          String samhsaAllowedCertificateAliasesJson,
+      @Qualifier("serverTrustStore") KeyStore keyStore) {
+    super();
+    Gson deserializer = new Gson();
```

Comment:
Maybe worth discussing if we want to standardize on a library for parsing json. Elsewhere we use Jackson.

Reply:
Sure, I went ahead and switched it to Jackson.